test/algodclientv2_test.go

```diff
@@ -243,6 +256,16 @@ func weMakeAnAccountApplicationInformationCallAgainstAccountApplicationID(accoun
 	return nil
 }
 
+func withClient(f func(client algod.Client)) error {
```

Comment:
```suggestion
func withClientAndMockServer(f func(client algod.Client)) error {
```

nit: suggest a name that clarifies to a new user of the codebase (such as me) that the request will be made against `mockServer.URL`

Reply:
I think I would refactor this part of code, so we don't have to declare a local function under cucumber step, and the refactoring would take this `withClient` function out.